docker-compose.yml

```diff
@@ -166,6 +167,7 @@ services:
       - IPFS_GATEWAY_URL=http://ipfs-proxy:9999
       - IPFS_API_URL=http://ipfs-proxy:9999
       - ENVKEY=
+      - ENABLE_EVENTCACHE_DB=true
```

Comment:
If both the listener and graphql use event-cache, will that cause double writes to the event table ?

Reply:
This is an excellent point.  I don't think it ever showed problems in my docker testing, but it's a serious concern.  I think this is still safe to merge and deploy since only one prod system has it enabled and since `PostgreSQLBackend` uses upsert functionality, it shouldn't cause any conflict.  

But that's a waste of resources and there should be some work done to either disable writes on some of the EC consumers, or add some kind of auto-locking feature that prevents more than one instance of EC to write. Probably the former, since I think it'll be a conscious decision of ours to make listener do the writes.

Good catch.  This didn't even cross my mind.

Reply:
Maybe to start with a simple solution, we could control this via an environment variable. Something like process.env.ENABLE_EVENT_CACHE_DB_WRITE which we would set to true only on either the graphql server or the listener. Then later if needed we can have a lock table or perhaps even move events writing to a completely separate service.